tcms/testcases/tests/test_views.py

```diff
@@ -228,6 +232,95 @@ def _assertTestCase(self, test_case):
         self.assertEqual(test_case.notes, self.notes)
 
 
+class TestNewCasePermission(PermissionsTestCase):
+    permission_label = 'testcases.add_testcase'
+    http_method_names = ['get', 'post']
+    url = reverse('testcases-new')
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.tester = UserFactory()
+        cls.case_status_confirmed = TestCaseStatus.objects.get(name='CONFIRMED')
+        cls.case_status_proposed = TestCaseStatus.objects.get(name='PROPOSED')
+
+        cls.product = ProductFactory()
+        cls.version = VersionFactory(value='0.1', product=cls.product)
+
+        cls.plan = TestPlanFactory(
+            author=cls.tester,
+            product=cls.product,
+            product_version=cls.version)
+
+        cls.case = TestCaseFactory(
+            author=cls.tester,
+            default_tester=None,
+            reviewer=cls.tester,
+            case_status=cls.case_status_confirmed,
+            plan=[cls.plan])
+        cls.case.save()  # will generate history object
+        cls.summary = 'summary'
+
+        cls.post_data = {
+            'author': cls.tester.pk,
+            'summary': cls.summary,
+            'default_tester': cls.tester.pk,
+            'product': cls.case.category.product.pk,
+            'category': cls.case.category.pk,
+            'case_status': cls.case_status_confirmed.pk,
+            'priority': cls.case.priority.pk,
```

Comment:
The values for most of these fields are not used anywhere outside of the setup method so you don't need them as class attributes.